src/HDInsight/HDInsight/ManagementCommands/NewAzureHDInsightClusterCommand.cs

```diff
@@ -664,20 +665,20 @@ private Guid GetApplicationId(Guid applicationId)
                 return applicationId;
             }
 
-            GraphRbacManagementClient graphClient = AzureSession.Instance.ClientFactory.CreateArmClient<GraphRbacManagementClient>(
+            MicrosoftGraphClient graphClient = AzureSession.Instance.ClientFactory.CreateArmClient<MicrosoftGraphClient>(
                 DefaultProfile.DefaultContext, AzureEnvironment.Endpoint.Graph);
 
             graphClient.TenantID = DefaultProfile.DefaultContext.Tenant.Id.ToString();
 
-            Microsoft.Azure.Graph.RBAC.Version1_6.Models.ServicePrincipal sp = null;
+            MicrosoftGraphServicePrincipal sp = null;
             try
             {
-                sp = graphClient.ServicePrincipals.Get(ObjectId.ToString());
+                sp = graphClient.ServicePrincipals.GetServicePrincipalWithHttpMessagesAsync(ObjectId.ToString()).Result.Body;//graphClient.ServicePrincipals.Get(ObjectId.ToString());
             }
-            catch (Microsoft.Azure.Graph.RBAC.Version1_6.Models.GraphErrorException e)
+            catch (Exception e)
             {
                 string errorMessage = e.Message + ". Please specify Application Id explicitly by providing ApplicationId parameter and retry.";
-                throw new Microsoft.Azure.Graph.RBAC.Version1_6.Models.GraphErrorException(errorMessage);
+                throw new Exception(errorMessage);
```

Comment:
Please use [AzPSArgumentException](https://github.com/Azure/azure-powershell-common/blob/main/src/Common/Exceptions/AzPSArgumentException.cs) so we can get detailed telemetry data.

Reply:
Updated